### DISCUSSION TITLE
SameSite cookie attribute

### DISCUSSION THREAD

**Discussion by jcdevil** [encode/httpx#2168] [Q&A]
Hello.
Using `httpx` in some functional tests (working with `http2`), I'd like to know what is the behaviour of httpx client in regards of the SameSite cookie attribute.
Making ref to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite
Does `httpx` behave like most recent browsers (ie implementing / managing the `Lax`, `None` and `Strict` values) ?
Thx for your help and for that wonderful client library.

**Comment by aikfiend** [comment]
I got a similar task today, it looks like `Cookie` and `CookieJar` from `http.cookiejar` [are being used](https://github.com/encode/httpx/blob/beb501fc28cb24ef7cd584159bc6ac55d2244dea/httpx/_models.py#L9) under the hood and they have corresponding interfaces `get_nonstandard_attr()` and `set_nonstandard_attr()` to get and set SameSite cookie. Inspect the code, nonstandard cookies will be inside "protected" `_rest` field.
